plugins/integrity_plugin.py

```diff
@@ -19,28 +20,48 @@ def on_run_begin(self, pipeline: Pipeline) -> Pipeline:
 
 
 def check_integrity(pipeline: Pipeline) -> bool:
-    hierarchy = list(reversed(flatten(pipeline.children())))
-    previous_block = hierarchy[0]
-    for item in hierarchy[1:]:
-        if isinstance(item, Pipeline):
-            continue
-        if isinstance(previous_block.inputTypes, List):
-            if (
-                item.outputType not in previous_block.inputTypes
-                and DataType.Any not in previous_block.inputTypes
-            ):
-                print(
-                    f"{item.id}'s output type is {item.outputType} and not {previous_block.inputTypes} which {previous_block.id} requires"
-                )
-                return False
-        else:
-            if (
-                item.outputType != previous_block.inputTypes
-                and previous_block.inputTypes is not DataType.Any
-            ):
-                print(
-                    f"{item.id}'s output type is {item.outputType} and not {previous_block.inputTypes} which {previous_block.id} requires"
-                )
+    hierarchy = pipeline.get_hierarchy()
```

Comment:
I re-wrote check_integrity to use the non-flat Hierarchy object, otherwise it was impossible to type check Concatenate objects properly!